controller/linodeobjectstoragebucket_controller.go

```diff
@@ -117,15 +115,6 @@ func (r *LinodeObjectStorageBucketReconciler) reconcile(ctx context.Context, bSc
 			reterr = err
 		}
 	}()
-
-	if !bScope.Bucket.DeletionTimestamp.IsZero() {
-		return res, r.reconcileDelete(bScope)
-	}
-
-	if err := bScope.AddFinalizer(ctx); err != nil {
-		return res, err
-	}
-
 	if err := r.reconcileApply(ctx, bScope); err != nil {
```

Comment:
We should still check if `!bScope.Bucket.DeletionTimestamp.IsZero()` and return `res, nil` to avoid invoking `r.reconcileApply` when the resource is deleted.

Reply:
Alternatively, we could implement Predicate's Delete function to always return false to ignore all deletions. Down at Line 174-
```go
		WithEventFilter(predicate.And(
			predicate.Funcs{DeleteFunc: func(event.DeleteEvent) bool { return false }},
			predicates.ResourceHasFilterLabel(mgr.GetLogger(), r.WatchFilterValue),
			predicate.GenerationChangedPredicate{},
		)).
```

Reply:
@bcm820 I checked a few times, even without your suggested predicate changes, the reconcile loop is not triggered at all on Deletion. I checked a few times and added logger statements while debugging as well. My understanding was, previously it was called due to an update event on the Deletiontimestamp and then reconciled. Is my understanding correct? Without finalizers, isn't the resource sent for deletion immediately?

Reply:
I think Brian the Reconcile is triggered but the error gets caught here `if err := r.TracedClient().Get(ctx, req.NamespacedName, objectStorageBucket); err != nil` and will never reach the reconcile apply. Found this as part of an article. **But after an object is deleted and the reconciler is called because of that delete, the call r.Get(ctx, req.NamespacedName, &obj) will fail with a not found err because the object is immediately deleted internally.** Do we still need the predicate to not initiate a reconciliation event during Deletion?

Reply:
Ah right, it doesn't find the object because it gets deleted right away. That's a pretty safe guarantee, and so filtering out deletes might be an over-optimization. I'd say let's not worry about the predicate at this time.